cache.go

```diff
@@ -0,0 +1,155 @@
+package sqlair
+
+import (
+	"context"
+	"database/sql"
+	"runtime"
+	"sync"
+	"sync/atomic"
+
+	"github.com/canonical/sqlair/internal/expr"
+)
+
+// stmtIDCount and dbIDCount are global variables to used to generate unique
+// cache IDs.
+var stmtIDCount int64
+var dbIDCount int64
+
+type dbID = int64
```

Comment:
I wouldn't bother creating types for these.

Reply:
Removed.